File: orientation-post-baccalauriat-BackEnd/src/main/java/com/biorbac/model/Student.java
package com.biorbac.model;

import com.biorbac.enums.BacType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Entity
@Getter
@Setter
public class Student extends User {

    private LocalDate dateOfBirth;
    private double bacScore;
    private String location;

    @Enumerated(EnumType.STRING)
    private BacType bacType;


    @OneToMany(mappedBy = "student")
    private List<Review> reviews;


}

File: orientation-post-baccalauriat-FrontEnd/src/app/core/model/student.ts
import {User} from "./user";
import {BacType} from "../enums/bac-type";
import {Institution} from "./institution";
import {Review} from "./review";


export interface Student extends User {
  dateOfBirth: Date;
  bacScore: number;
  location: string;
  bacType: BacType;
  institutions: Institution[];
  reviews: Review[];
}

